### PR TITLE
CommitState::WriteCatalogEntry refactor

### PR DESCRIPTION
**Pull request by carlopi**
Now most entries go also through the alter codepath.

Problem found with @Tishj, and solution discussed also with @Mytherin, added both as reviewers.

Also adding the original test with changed settings so that it will fail (without this changes) also with default configurations.

### CODE REVIEW

**Comment by Tishj** [comment]
This fixes https://github.com/duckdblabs/duckdb-internal/issues/1225

To describe the problem in some more detail:
These catalog types have never had the ability to be ALTERed, so the codepath to deal with that didn't exist yet
Instead it would use the `Create<type>` method on the WAL, which would (seemingly) silently fail because the entry already exists

Now it properly uses WriteAlter to update the entry

File: src/transaction/commit_state.cpp [hunk 1]
```diff
@@ -57,18 +63,61 @@ void CommitState::WriteCatalogEntry(CatalogEntry &entry, data_ptr_t dataptr) {
...
+				log->WriteCreateType(parent.Cast<TypeCatalogEntry>());
+				break;
+			case CatalogType::MACRO_ENTRY:
+				log->WriteCreateMacro(parent.Cast<ScalarMacroCatalogEntry>());
+				break;
+			case CatalogType::TABLE_MACRO_ENTRY:
+				log->WriteCreateTableMacro(parent.Cast<TableMacroCatalogEntry>());
+				break;
+			default:
+				throw InternalException("Don't know how to drop this type!");
```
Comment (Tishj) [inline src/transaction/commit_state.cpp:119]: nit: `Don't know how to create this type!`

File: src/transaction/commit_state.cpp [hunk 2]
```diff
@@ -57,18 +63,61 @@ void CommitState::WriteCatalogEntry(CatalogEntry &entry, data_ptr_t dataptr) {
...
+			case CatalogType::VIEW_ENTRY:
+			case CatalogType::INDEX_ENTRY:
+			case CatalogType::SEQUENCE_ENTRY:
+			case CatalogType::TYPE_ENTRY:
+			case CatalogType::MACRO_ENTRY:
+			case CatalogType::TABLE_MACRO_ENTRY:
+				(void)column_name;
+				break;
+			default:
+				throw InternalException("Don't know how to drop this type!");
```
Comment (Tishj) [inline src/transaction/commit_state.cpp:85]: `Don't know how to alter this type!`

**Review by Tishj — APPROVED** [review]
Looks good to me, thanks!

**Comment by Mytherin** [comment]
Thanks!

**Comment by carlopi** [comment]
Thanks you both for the support! I will fix the comments in a follow up.

**Pull request merged**